src/jquery.uls.core.js

```diff
@@ -225,30 +264,26 @@
 				this.$menu.on( 'keydown', $.proxy( this.keypress, this ) );
 			}
 
-			languagesCount = Object.keys( this.options.languages ).length;
-			lcd = this.$resultsView.lcd( {
-				languages: this.languages,
-				columns: columnsOptions[ this.getMenuWidth() ],
-
-				quickList: languagesCount > 12 ? this.options.quickList : [],
-				clickhandler: $.proxy( this.select, this ),
-				source: this.$languageFilter,
-				showRegions: this.options.showRegions,
-				languageDecorator: this.options.languageDecorator,
-				noResultsTemplate: this.options.noResultsTemplate
-			} ).data( 'lcd' );
+			this.createLanguageFilter();
 
-			this.$languageFilter.languagefilter( {
-				$target: lcd,
-				languages: this.languages,
-				searchAPI: this.options.searchAPI,
-				onSelect: $.proxy( this.select, this )
-			} );
-
-			this.$languageFilter.on( 'noresults.uls', $.proxy( lcd.noResults, lcd ) );
 			this.$languageFilter.on( 'resultsfound.uls', $.proxy( this.success, this ) );
 
 			$( 'html' ).click( $.proxy( this.cancel, this ) );
+			$( window ).resize( $.debounce( 250, this.resize.bind( this ) ) );
```

Comment:
$.debounce is not defined. have a look at delay method in jquery.uls.langaugefilter.js

Reply:
What happens if window is resized while user has typed a search query and we recreate LCD? Will we just throw away the search results?

Reply:
Debounce function added in 90d4808.

Niklas: LCD is recreated only if not showing. Adding responsiveness while ULS is visible would require some changes in core logic, but that is something I consider as future feature.